### ISSUE TITLE
Report unnecessary else after continue and break

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#2327]
Similarly as reporting `no-else-return`, it would be nice to also report unnecessary `else` after `continue` and `break`.

```python
def something():
    for i in range(10):
        if i < 3:
            continue  # or break
        else:  # Unnecessary else after "continue" or "break"
            print(i)
```

**Comment by sushobhit27** [comment]
@scop what if there are statements in for loop after if-else block also?

**Comment by scop** [comment]
I'm not sure I understand the question -- could you clarify it by posting some sample code? Anyway the `else` would still be necessary in this example:

```python
def something():
    for i in range(10):
        if i < 3:
            continue  # or break
        else:  # Unnecessary else after "continue" or "break"
            print(i)
        # some code goes here
        # more code goes here
        # even more code...
```

**Comment by sushobhit27** [comment]
yes I meant the same example... in this case this message should not be emitted.

**Comment by scop** [comment]
Um, I don't follow, sure it should be emitted then too? The behavior is the same as in this fixed code without the `else`, no?

```python
def something():
    for i in range(10):
        if i < 3:
            continue  # or break
        print(i)
        # some code goes here
        # more code goes here
        # even more code...

**Comment by sushobhit27** [comment]
@scop sorry :) error should be emitted in this case, both cases are same.

**Comment by PCManticore** [comment]
Yes, this could be a nice a little refactoring message.
